unit_tests/testLinearSolver/testLinearSolverParallel.f90

```diff
@@ -0,0 +1,177 @@
+  CALL pList%add('LinearSolverType->A->MatrixType->nnz',33)
+  CALL pList%add('LinearSolverType->x->VectorType->n',9)
+  CALL pList%add('LinearSolverType->b->VectorType->n',9)
+  CALL pList%validate(pList)
+  CALL thisLS%init(pList)
+
+  !A =  4    -1     0    -1     0     0     0     0     0
+  !    -1     4    -1     0    -1     0     0     0     0
+  !     0    -1     4     0     0    -1     0     0     0
+  !    -1     0     0     4    -1     0    -1     0     0
+  !     0    -1     0    -1     4    -1     0    -1     0
+  !     0     0    -1     0    -1     4     0     0    -1
+  !     0     0     0    -1     0     0     4    -1     0
+  !     0     0     0     0    -1     0    -1     4    -1
+  !     0     0     0     0     0    -1     0    -1     4
+  SELECTTYPE(A => thisLS%A); TYPE IS(DistributedBandedMatrixType)
+    CALL A%set(1,1, 4.0_SRK)
+    CALL A%set(1,2,-1.0_SRK)
+    CALL A%set(1,4,-1.0_SRK)
+    CALL A%set(2,1,-1.0_SRK)
+    CALL A%set(2,2, 4.0_SRK)
+    CALL A%set(2,3,-1.0_SRK)
+    CALL A%set(2,5,-1.0_SRK)
+    CALL A%set(3,2,-1.0_SRK)
+    CALL A%set(3,3, 4.0_SRK)
+    CALL A%set(3,6,-1.0_SRK)
+    CALL A%set(4,1,-1.0_SRK)
+    CALL A%set(4,4, 4.0_SRK)
+    CALL A%set(4,5,-1.0_SRK)
+    CALL A%set(4,7,-1.0_SRK)
+    CALL A%set(5,2,-1.0_SRK)
+    CALL A%set(5,4,-1.0_SRK)
+    CALL A%set(5,5, 4.0_SRK)
+    CALL A%set(5,6,-1.0_SRK)
+    CALL A%set(5,8,-1.0_SRK)
+    CALL A%set(6,3,-1.0_SRK)
+    CALL A%set(6,5,-1.0_SRK)
+    CALL A%set(6,6, 4.0_SRK)
+    CALL A%set(6,9,-1.0_SRK)
+    CALL A%set(7,4,-1.0_SRK)
+    CALL A%set(7,7, 4.0_SRK)
+    CALL A%set(7,8,-1.0_SRK)
+    CALL A%set(8,5,-1.0_SRK)
+    CALL A%set(8,7,-1.0_SRK)
+    CALL A%set(8,8, 4.0_SRK)
+    CALL A%set(8,9,-1.0_SRK)
+    CALL A%set(9,6,-1.0_SRK)
+    CALL A%set(9,8,-1.0_SRK)
+    CALL A%set(9,9, 4.0_SRK)
+    CALL A%assemble()
+  ENDSELECT
+
+  ! Use default initial guess (all ones)
+  SELECTTYPE(x => thisLS%x); TYPE IS(NativeDistributedVectorType)
+    x%b = 1.0_SRK
+  ENDSELECT
+
+  SELECTTYPE(b => thisLS%b); TYPE IS(NativeDistributedVectorType)
+    b%b = 1.0_SRK
+  ENDSELECT
+
+  !set iterations and convergence information and build/set M
+  SELECTTYPE(thisLS); TYPE IS(LinearSolverType_Iterative)
+    thisLS%hasX0 = .TRUE.
+    CALL thisLS%setConv(2,1.0E-6_SRK,1.0E-6_SRK,20,4)
+  ENDSELECT
+
+
+  !solve it
+  CALL thisLS%solve()
+
+  !Store expected solution (from MATLAB) in B
+  IF(rank == 0) THEN
+    ALLOCATE(dummyvec(5))
+    ALLOCATE(thisB(5))
+    thisB(:) = (/0.6875_SRK,0.875_SRK,0.6875_SRK,0.875_SRK,1.125_SRK/)
+    CALL thisLS%X%get(dummyvec)
+    ASSERT(ALL(SOFTEQ(thisB(:),dummyvec(:),1.0E-05_SRK)), 'CALL Iterative%solve() -GMRES FAILED!')
+  ELSE
+    ALLOCATE(dummyvec(4))
+    ALLOCATE(thisB(4))
+    thisB(:)=(/0.875_SRK,0.6875_SRK,0.875_SRK,0.6875_SRK/)
+    CALL thisLS%X%get(dummyvec)
+    ASSERT(ALL(SOFTEQ(thisB(:),dummyvec(:),1.0E-05_SRK)), 'CALL Iterative%solve() -GMRES FAILED!')
+  ENDIF
```

Comment:
```suggestion
  IF(rank == 0) THEN
    thisB(:) = (/0.6875_SRK,0.875_SRK,0.6875_SRK,0.875_SRK,1.125_SRK/)
  ELSE
    thisB(:)=(/0.875_SRK,0.6875_SRK,0.875_SRK,0.6875_SRK/)
  ENDIF
  CALL thisLS%X%get(dummyvec)
  ASSERT(ALL(SOFTEQ(thisB(:),dummyvec(:),1.0E-05_SRK)), 'CALL Iterative%solve() -GMRES FAILED!')
```

Reply:
This is assuming that thisLS%get doesn't require a pre-allocated array

Reply:
Actually the %get function is a member of the VectorTypes class, which is specified to work only on pre-allocated arrays. The interface %get handles accessing a single element, range of elements, or a subset of elements, so behavior in the case of an unallocated input is not well-defined.

Reply:
So first things first, this is a super trivial request in the grand scheme of things. The bigger thing,
```
SUBROUTINE getAll_PETScVectorType(thisVector,getval,ierr)
  CLASS(PETScVectorType),INTENT(INOUT) :: thisVector     
  REAL(SRK),INTENT(OUT) :: getval(:)                     
```
I am pretty sure that using the allocation status/size of an `intent(out)` is undefined. At a minimum it is probably bad form, I invite others to comment/correct as necessary.

Reply:
You can only check allocation status on an `ALLOCATABLE` dummy argument.  That dummy argument should probably be allocatable, but that's very much outside the scope of this PR.

I like the spirit of your suggestion but I'm not sure how much of it we can use.  You good just resolving this as is?

Reply:
In light of the constraints with existing code, I updated the suggestion to something of a middle ground. It is essentially take it or leave it, just be aware of chances to streamline that sort of logic and make things easier for follow on development.